OpenRA.Mods.Common/Traits/UpgradesIcons.cs

```diff
@@ -0,0 +1,61 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Attach this to the player actor. When attached, enables all actors possessing the UpgradesWhenCreated" +
+		"trait to have their production queue icons changed from Tooltip.Icon to Tooltip.IconUpgraded. The icon change" +
+		"occurs when UpgradesWhenCreated.Prerequisites are met.")]
+	public class UpgradesIconsInfo : ITraitInfo
```

Comment:
Make this implement `Requires<TechTreeInfo>` to avoid initialization-order issues.